sql/connect/server/src/main/scala/org/apache/spark/sql/connect/ml/MLCache.scala

```diff
@@ -243,7 +242,7 @@ private[connect] class MLCache(sessionHolder: SessionHolder) extends Logging {
     val size = cachedModel.size()
     cachedModel.clear()
     if (getMemoryControlEnabled) {
-      FileUtils.cleanDirectory(new File(offloadedModelsDir.toString))
+      SparkFileUtils.cleanDirectory(new File(offloadedModelsDir.toString))
```

Comment:
@dongjoon-hyun I would like to further confirm whether it's acceptable to directly use `JavaUtils#deleteRecursively(java.io.File)` here. It seems the difference is that `JavaUtils#deleteRecursively(java.io.File)` will delete the root directory, whereas `cleanDirectory` will not. Additionally, `cleanDirectory` will clean up symbolic links, but `deleteRecursively` will not.

Reply:
I'm not sure whether retaining the root directory is important.

Reply:
I left a comment to the authors here.
- https://github.com/apache/spark/pull/50752/files#r2250002745